lib/rex/post/meterpreter/packet.rb

```diff
@@ -665,6 +665,44 @@ def initialize(type = nil, method = nil)
     end
   end
 
+  #
+  # Override the function that creates the raw byte stream for
+  # sending so that it generates an XOR key, uses it to scramble
+  # the serialized TLV content, and then returns the key plus the
+  # scrambled data as the payload.
+  #
+  def to_r
```

Comment:
I may just be overly paranoid, but I felt like it was a good thing to guarantee that every byte changed. That's really the only reasoning.